src/field_theory/finite.lean

```diff
@@ -129,3 +130,53 @@ calc a ^ (fintype.card α - 1) = (units.mk0 a ha ^ (fintype.card α - 1) : units
   ... = 1 : by rw [← card_units, pow_card_eq_one]; refl
 
 end finite_field
+
+namespace zmodp
+
+open finite_field
+
+lemma sum_two_squares_of_odd {p : ℕ} (hp : p.prime) (hp2 : p % 2 = 1) (x : zmodp p hp) :
```

Comment:
This doesn't need `hp2`, right? If `p % 2 = 0`, then I can take `a = b = 0` or (`a = 0` and `b = 1`). Taking care of this extra case means you don't need the lemma coming after this one.
Similarly, in other lemmas in this file, I think you can remove oddness assumptions.

Reply:
Yes. I've no idea what came over me when I did that.